src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -4491,6 +4487,7 @@ public void warn(LintCategory lint) {
             switch (lint) {
                 case UNCHECKED:
                     Check.this.warnUnchecked(pos(), Warnings.ProbFoundReq(diags.fragment(uncheckedKey), found, expected));
+                    this.warned = true;
```

Comment:
Odd - this field is clearly meant to be set in this method, but it is not in mainline. Maybe better to set the field only once, after the switch?